Automattic/wp-calypso
# Jetpack UX: Add xml and rest api error handling

Related to https://github.com/Automattic/dotcom-forge/issues/3241

## Proposed Changes

This pr add support for jetpack XML-RPC and REST API connection is

## Testing Instructions

### Test 1: REST API errors
1. Apply D119857-code
2. Create a new WoA site.
3. Enable ssh.
4. Visit eg `/posts` page
5. Login to the site and add the following to your `wp-config`
```
if ( ! empty( $_GET['rest_route'] ) ) {
	http_response_code( 404 );
	die('not found');
}
```
6. Ensure that you see the following error:
![rest_api_jetpack](https://github.com/Automattic/wp-calypso/assets/497103/ce4debbd-b251-4497-a65e-c39fc29ef14b)

### Test 2: XML-RPC errors
1. Apply D119857-code
2. Create a new WoA site.
3. Enable ssh.
4. Visit eg `/posts` page
5. Login to the site and add the following to your `wp-config`
```
if ( ! empty( $_GET['for'] ) && $_GET['for'] === 'jetpack' ) {
	http_response_code( 404 );
	die('not found');
}
```
6. Ensure that you see the following error:
![xml-rpc_jetpack](https://github.com/Automattic/wp-calypso/assets/497103/957f6cb4-f5a1-4517-9e61-bce68ff1fa89)



## Pre-merge Checklist
